src/juju/api.ts

```diff
@@ -457,14 +469,12 @@ export async function fetchControllerList(
   @param conn The controller connection instance.
 */
 export function disableControllerUUIDMasking(conn: ConnectionWithFacades) {
-  return new Promise<void>(async (resolve, reject) => {
+  return new Promise<void>((resolve, reject) => {
     if (conn?.facades?.jimM) {
-      try {
-        await conn.facades.jimM.disableControllerUUIDMasking();
-        resolve();
-      } catch (e) {
-        reject();
-      }
+      conn.facades.jimM
+        .disableControllerUUIDMasking()
+        .then(() => resolve())
+        .catch((e) => reject(new Error()));
```

Comment:
I think the error that is thrown here should have a message, maybe something like:

```suggestion
        .catch((e) => reject(new Error("Unable to disabled controller UUID masking.", e)));
```

Reply:
Good suggestion! Changed it in my latest commit.